### ISSUE TITLE
Gaps in moving to Lens

### ISSUE THREAD

**Issue by andresrc** [elastic/integrations#265]
The goal of this issue is to collect the different issues / gaps that we might find when building new dashboards to Lens or migrating existing ones.

Please add comments withe the issues you find, they will be consolidated in the main comment.

**Comment by kaiyan-sheng** [comment]
It's very convenient to use group by `terms` in lens but not group by `filter` like this shown in TSVB:
<img width="1097" alt="Screen Shot 2020-09-14 at 10 44 19 AM" src="https://user-images.githubusercontent.com/14081635/93114136-74d6d980-f677-11ea-8280-22d413caeb20.png">

**Comment by ChrsMark** [comment]
Lens does not provide support for percentile aggregation:
![96571669-8ba2c880-12d4-11eb-87f3-9e3e050bbac3](https://user-images.githubusercontent.com/11754898/97552388-df03cd80-19dc-11eb-8436-edcc70adda0e.png)

TSVB is the way to go when one want to visualise histograms.

**Comment by AlonaNadler** [comment]
@kaiyan-sheng filter aggregation is not supported in Lens in 7.10
please let me know if you managed to create what you wanted

**Comment by kaiyan-sheng** [comment]
Thanks @AlonaNadler I used TSVB for this use case. Do you know if we are planning to add filter aggregation into Lens soon?

**Comment by AlonaNadler** [comment]
Sorry I had a typo. Filter aggregation **is** supported in Lens in 7.10

**Comment by kaiyan-sheng** [comment]
This is not a gap but more of an enhancement request: when we create a lens visualization and plot a field with break down by `container.id`, is it possible to add a display name field to be `container.name` instead? For this case, `container.id` is just a random string that's not human readable but `container.name` is actually a meaningful keyword. But `container.name` can be the same for different containers running in different clusters so we have to plot with `container.id` instead. Similar use case for `host.id` and `host.name` as well.

**Comment by mtojek** [comment]
Currently Kibana doesn't support Lens type in fleet (/cc @ph).

Also, I've just opened a PR to enable it in the package-spec: https://github.com/elastic/package-spec/pull/94

**Comment by ph** [comment]
We aren't support lens assets installation yet, I've created this on our side https://github.com/elastic/kibana/issues/85428 , I've marked it as 77.12 because I think its low effort to add and is required to use lens in integration.

**Comment by skh** [comment]
PR open for EPM support for Lens assets ( https://github.com/elastic/kibana/issues/85428 ): https://github.com/elastic/kibana/issues/85428

**Comment by ph** [comment]
Everything is done in the package integration to support LEns FYI @masci

**Comment by kaiyan-sheng** [comment]
Not sure if this is a gap for `Lens` or it is by design. We actually use a lot of `Controls` visualization in dashboards to allow users to filter on cloud account ID, regions, zones and etc. Currently, we are using `Controls` visualization for this and `Controls` is still marked as experimental(in 7.10). Will this be a part of `Lens` eventually? 
<img width="317" alt="Screen Shot 2021-01-25 at 8 59 31 AM" src="https://user-images.githubusercontent.com/14081635/105730506-ad22da00-5eeb-11eb-8096-02fdb278f5b3.png">

**Comment by timductive** [comment]
Hi @kaiyan-sheng it looks like this is Dashboard Input Controls, this is on our radar to bring out of experimental and GA sometime soon. This is not part of Lens though, this will likely stay a part of the Dashboard. You can follow along here: https://github.com/elastic/kibana/issues/17231

**Comment by kaiyan-sheng** [comment]
Awesome! @timductive Thank you for the info!

**Comment by wylieconlon** [comment]
Lens will support the Histogram data type in 7.12 for all the standard numeric operations except Cardinality.

**Comment by rayafratkina** [comment]
Looks like at this point all Lens gaps have been resolved and the only remaining issue is around Input controls. Should we close this issue and start tracking Dashboard-level dependencies (including Input controls)?
See https://github.com/elastic/kibana/issues/59821 for the input controls plan.

**Comment by matschaffer** [comment]
Did something like TSVBs "Positive Rate" land somewhere? We use that a lot in Cloud (often for plotting deeper parts of stack monitoring data).

**Comment by wylieconlon** [comment]
Yes, counter rate was merged for 7.12. It's also called counter rate in TSVB now: https://www.elastic.co/guide/en/kibana/current/whats-new.html#lens-7-12

**Comment by matschaffer** [comment]
Oh, and share-without-save was another one on my nice-to-haves list, though that's more just for general workflow (so we can paste to slack) rather than something that'd block an integration moving over

**Comment by wylieconlon** [comment]
@matschaffer It sounds like you have other items that you're asking for, can you list all of them? We have never discussed sharing without saving before, so I think we will need to do some investigation into it.

**Comment by tbragin** [comment]
@drewpost @sorantis @mukeshelastic Are you comfortable with this issue being closed? I am not sure if this is the only place we are tracking the delta between Lens capabilities today and what we need for Observability Integrations & Data Exploration use cases.

**Comment by iankoetter** [comment]
@Drew0522 I think issue #75316 was overlooked here potentially. I currently cannot find a way to share a link to a viz created explicitly in lens. This is different from other visualization user experiences with the existing 'share' submenu options.

**Comment by Drew0522** [comment]
@iankoetter I think you tagged the wrong Drew :)

**Comment by ruflin** [comment]
Checked off the last box in the issue description around "dashboard input controls" as it is available now: https://www.elastic.co/guide/en/kibana/master/add-controls.html Going to close this.

@andrewctate There is one open issue that is discussed above which is https://github.com/elastic/kibana/issues/75316 Would be great if could have a look at this in the context of Lens adoption.

**Comment by andrewctate** [comment]
@ruflin https://github.com/elastic/kibana/issues/75316 is on our immediate roadmap. Planned for 8.7.
